hair_salon/data/ir_actions_server.xml

```diff
@@ -0,0 +1,67 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="break_booking_server_action" model="ir.actions.server">
+        <field name="code"><![CDATA[
+                env['calendar.event'].search([('x_parent_id', '=', record.id)]).unlink()
+                        ]]></field>
+        <field name="model_id" ref="calendar.model_calendar_event"/>
+        <field name="state">code</field>
+        <field name="name">Execute Code</field>
+    </record>
+    <record id="delete_subpart_main_calendar_event_server_action" model="ir.actions.server">
+        <field name="child_ids" eval="[(6, 0, [ref('break_booking_server_action')])]"/>
+        <field name="code"><![CDATA[parent_record = env['calendar.event'].search([('id', '=', record.x_parent_id.id)])
+if parent_record.x_appointment_part_2.id == record.id:
+        parent_record.x_appointment_part_2.unlink()
+else:
+        parent_record.x_appointment_part_1.unlink()
+parent_record["active"] = False]]></field>
+        <field name="model_id" ref="calendar.model_calendar_event"/>
+        <field name="state">code</field>
+        <field name="name">Execute Code</field>
+    </record>
+    <record id="delete_subpart_calendar_event_server_action" model="ir.actions.server">
+        <field name="code"><![CDATA[x_break_start = record.x_break_start
+x_break_end = record.x_break_end
+appointment_duration = record.duration
+
+start_time = record.start
+end_time = record.stop
+if x_break_end > 0 and x_break_start > 0 :
+        event2_vals = {
+                'name': "[PART 1]" + record.name ,
+                'start': start_time,
+                'stop': start_time + datetime.timedelta(hours=x_break_start),
+                'allday': record.allday,
+                'user_id': record.user_id.id,
+                'x_parent_id': record.id,
+                'alarm_ids' : [],
+                'partner_ids': [record.partner_ids[0].id],
+                'appointment_type_id':record.appointment_type_id.id,
+                
+        }
```

Comment:
All theses fields would need to be changed except maybe the user_id and the appointment_type_id. 
But we will need to duplicate so +1 line. What would be the advantage of shorter it? Because this is for format but could be written in 1 line.